src/vcpkg/vcpkgcmdarguments.cpp

```diff
@@ -717,6 +769,18 @@ namespace vcpkg
         from_env(get_env, ASSET_SOURCES_ENV, asset_sources_template_env);
         from_env(get_env, REGISTRIES_CACHE_DIR_ENV, registries_cache_dir);
 
+        // detect whether we are running in a CI environment
+        for (auto&& ci_env_var : get_known_ci_variables())
```

Comment:
Do we want a documented `VCPKG_NO_CI` env var to disable this check?

I can see us wanting to change behavior based on this in the future, so we should give users a way to opt-out.

Reply:
I like the idea so went ahead and added the opt-out variable.